graylog2-server/src/main/java/org/graylog2/rest/resources/streams/rules/StreamRuleResource.java

```diff
@@ -118,7 +118,7 @@ public SingleStreamRuleSummaryResponse update(@ApiParam(name = "streamid", value
                                                   @ApiParam(name = "JSON body", required = true)
                                                   @Valid @NotNull CreateStreamRuleRequest cr) throws NotFoundException, ValidationException {
         checkPermission(RestPermissions.STREAMS_EDIT, streamid);
-        checkNotDefaultStream(streamid, "Cannot update stream rules on default stream.");
+        checkNotEditable(streamid, "Cannot update stream rules on non-editable streams.");
```

Comment:
The rules are now disabled for non-editable stream 👍 However outputs can still be created / assigned.